lib/commands/install.ts

```diff
@@ -0,0 +1,83 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import path = require("path");
+
+export class InstallCommand implements ICommand {
+	private _projectFilePath: string;
+	
+	constructor(private $fs: IFileSystem,
+		private $errors: IErrors,
+		private $logger: ILogger,
+		private $options: IOptions,
+		private $injector: IInjector,
+		private $staticConfig: IStaticConfig) { }
+		
+	public enableHooks = false;
+		
+	public allowedParameters: ICommandParameter[] = [];
+	
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+			let projectFilePath = this.getProjectFilePath(args[0]);
+			let projectData = this.getProjectData(projectFilePath).wait();
+			let projectName = projectData.id.split(".")[2];
+			
+			this.$injector.resolve("projectService").createProject(projectName).wait(); 
+			
+			this.$options.path = path.join(this.$options.path || path.resolve("."), projectName);
+			
+			this.$logger.info("Adding platforms...");
+			
+			let $platformsData = this.$injector.resolve("platformsData");
+			let $platformService = this.$injector.resolve("platformService");
+			_.each($platformsData.platformsNames, platform => {
+				let platformData = $platformsData.getPlatformData(platform);
+				let frameworkPackageData = projectData[platformData.frameworkPackageName];
+				if(frameworkPackageData && frameworkPackageData.version) {
+					$platformService.addPlatforms([`${platform}@${frameworkPackageData.version}`]).wait();
+				}
+			}); 
+			
+		}).future<void>()();
+	}
+	
+	public canExecute(args: string[]): IFuture<boolean> {
+		return (() => {
+			if(!args[0]) {
+				this.$errors.failWithoutHelp("You must specify the path to package.json file.");	
+			}
+			
+			let projectFilePath = this.getProjectFilePath(args[0]);
+			if(!this.$fs.exists(projectFilePath).wait()) {
+				this.$errors.failWithoutHelp("The provided path doesn't contain package.json.");	
+			}
+			
+			let projectData = this.getProjectData(projectFilePath).wait();
+			if(!projectData) {
+				this.$errors.failWithoutHelp("Invalid project file. Verify that the specified package.json file contains a nativescript key and try again."); 
+			}
+			
+			return true;
+		}).future<boolean>()();
+	}
+	
+	private getProjectFilePath(providedPath: string): string {
+		if(!this._projectFilePath) {
+			providedPath = path.resolve(providedPath);
+			this._projectFilePath = path.basename(providedPath) === "package.json" ? providedPath : path.join(providedPath, "package.json");
+		}
+		
+		return this._projectFilePath;
+	}
+	
+	private getProjectData(projectFilePath: string): IFuture<any> {
+		return (() => {
+			let fileContent = this.$fs.readJson(projectFilePath).wait();
```

Comment:
as @teobugslayer pointed, it will be great if you can cache the projectData as well